portfolio/src/test/java/com/google/sps/GenreRecommenderTest.java

```diff
@@ -0,0 +1,135 @@
+package com.google.sps;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.sps.data.Book;
+import com.google.sps.data.BookReader;
+import com.google.sps.data.GenreRecommender;
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * This is a test class for BookReader.java
+ * It only tests the basic functionality for reading from a CSV file.
+ * It does not check that the results match those from the Books API.
+ */
+@RunWith(JUnit4.class)
+public final class GenreRecommenderTest {
+  private List<Book> bookList;
```

Comment:
nit: May be move this declaration within the test, as it is not user elsewhere. Global variables in test are more useful when they are used in setup/teardown before/after code, or are final/static variables.